packages/mollie-integration/src/mollie-payment-strategy.ts

```diff
@@ -264,34 +250,31 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
             await form.validate();
             await form.submit(payment);
 
-            return this._store.dispatch(this._orderActionCreator.loadCurrentOrder());
+            await this.paymentIntegrationService.loadCurrentOrder();
```

Comment:
I think we should add return here to save the prev behaviour:
```suggestion
            return await this.paymentIntegrationService.loadCurrentOrder();
```

Reply:
Or we should add else to this condition and place 
`await this.paymentIntegrationService.submitPayment(payment);` there.

The main reason why I am asking to do so is `submitPayment` method will be called in both variants, but in the prev implementation it was called only if the `_isHostedPaymentFormEnabled` returns false.